bbbc021_classification_analysis.py

```diff
@@ -0,0 +1,342 @@
+from sklearn.cross_validation import cross_val_predict
+from sklearn.model_selection import cross_val_score
+from sklearn.metrics import confusion_matrix
+
+import itertools
+import pandas as pd
+import os
+import re
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+PATH = "/home/draga/FIT2082/Data_CSVs/"
+OUT_PATH = "/home/draga/FIT2082/Data_Scatters/"
+pd.set_option('display.max_columns', None)
+
+def main():
+    h = 0.05  # step size in the mesh (.02 initially caused memory error)
+
+    all_accuracies = []
+    all_labels = []
+
+    # for each Data CSV available
+    for filename in os.listdir(PATH):
+        pattern = r"(Data2*)_*(.*)(_group).csv"
+
+        match = re.search(pattern, filename)
+        if match:
+            # load all data points from CSV
+            point_data_full = pd.read_csv(PATH + filename)
+
+            # split data into points with labelled and unlabelled MOA
+            no_nones = point_data_full.loc[point_data_full['group'] != 'NONE']
+            nones = pd.concat([point_data_full, no_nones]).drop_duplicates(keep=False)
+
+            # generate dummy integer values for the group column
+            group_dummies = pd.get_dummies(no_nones['group']).values.argmax(1)
+            no_nones['group.dummy'] = group_dummies
+
+            # get equal(ish) samples of each category
+            sample_sizes = get_sample_sizes(no_nones, 30)
+            no_nones = get_sampled_data(no_nones, group_dummies, sample_sizes)
+
+            # define x and y values for plotting and classifying
+            x_points = no_nones['x'].values
+            y_points = no_nones['y'].values
+            X = np.stack((x_points, y_points), axis=1)
+            y = no_nones['group.dummy'].values
+
+
+            cmap_light = ListedColormap(sns.color_palette('pastel', 13))
+
+            colours_bold = list(sns.color_palette('bright', 13))
+
+            # cross reference colour to group for plotting
+            cats = np.unique(no_nones['group'])
+            color_dict = dict(zip(cats, colours_bold))
+            colour_col = no_nones['group'].apply(lambda x: color_dict[x])
+
+            # find best model for this data
+            k, accuracies = get_best_k(X, y)
+
+            # track accuracies and labels for all k values
+            label = (match.group(2) if len(match.group(2)) != 0 else "PCA") + (
+            "_all_objects" if len(match.group(1)) != 5 else "_sampled")
+            all_labels.extend([label for _ in range(len(accuracies))])
+            all_accuracies.extend(accuracies)
+
+            # fit data
+            clf = neighbors.KNeighborsClassifier(k, weights='distance')
+            clf.fit(X, y)
+
+            # # plot confusion matrix
+            y_pred = cross_val_predict(clf, X, y, cv=5)
+            scores = cross_val_score(clf, X, y, cv=5)
+
+            cnf_matrix = confusion_matrix(y, y_pred)
+            np.set_printoptions(precision=2)
+
+            # Plot confusion matrix
```

Comment:
There are large blocks of commented out code here. Is that because they are incorrect/buggy (and should be deleted), or because they are optional (and should be made into functions instead)?

Reply:
They are optional, and should be functions instead

Reply:
Would you be able to bundle them up into functions then? Thanks for submitting the PR!

Reply:
Yep, can do, but not until Friday at the earliest sorry!

Reply:
There's no rush, you don't have to be sorry. It's certainly not holding anything else up, this is very self contained.